R/PlainTextParam.R

```diff
@@ -177,38 +182,36 @@ setMethod("storeResults",
 setMethod("loadResults",
           signature(object = "MsExperiment",
                     param = "PlainTextParam"),
-          function(object, param, spectraPath = character()){
+          function(object, param, spectraPath = character()) {
               fl <- file.path(param@path, "sample_data.txt")
-              if (file.exists(fl)){ # should i have a error if does not exist ?
-                  sd <- read.table(fl)
-                  rownames(sd) <- NULL #read.table force numbering of rownames
-              }
-              s <- loadResults(Spectra(), param)
+              if (!file.exists(fl))
+                  stop("No 'sample_data.txt' file found in the provided path.")
+              sd <- read.table(fl, sep = "\t")
+              rownames(sd) <- NULL #read.table force numbering of rownames
+              s <- loadResults(Spectra(), param, spectraPath = spectraPath)
               res <- MsExperiment(sampleData = sd, spectra = s)
               validObject(res)
               res
-              }
-)
+          })
 
 #' @rdname PlainTextParam
 setMethod("loadResults",
           signature(object = "XcmsExperiment",
                     param = "PlainTextParam"),
-          function(object, param, spectraPath){
-              res <- callNextMethod() #check if need to add spectraPath = spectraPath
+          function(object, param, spectraPath = character()) {
+              res <- callNextMethod()
               res <- .load_xcmsexperiment(res, path = param@path)
               validObject(res)
               res
-          }
-)
+          })
 
 #' @rdname PlainTextParam
 setMethod("storeResults", signature(object = "Spectra",
                                     param = "PlainTextParam"),
           function(object, param) {
               dir.create(path = param@path,
                          recursive = TRUE,
-                         showWarnings = TRUE)
+                         showWarnings = FALSE)
```

Comment:
I would disable the warnings, because a `storeResults,Spectra` will create the directory, save data there and then call `storeSpectra,MsBackendMzR` which will throw a warning because the directory was already created. So, we get these warnings even in a typical workflow - which might confuse users.